File: bindings/android/kotlin/com/anychat/sdk/AnyChatClient.kt
package com.anychat.sdk

/**
 * Kotlin wrapper around the SWIG-generated JNI bindings.
 * Provides a coroutine-friendly API for Android developers.
 */
class AnyChatClient(private val config: ClientConfig) {

    init {
        System.loadLibrary("anychat_android")
    }

    // TODO: delegate to SWIG-generated native methods
    external fun nativeConnect(): Long
    external fun nativeDisconnect(handle: Long)
    external fun nativeGetConnectionState(handle: Long): Int

    companion object {
        // Maps native ConnectionState int to Kotlin enum
        enum class State { DISCONNECTED, CONNECTING, CONNECTED, RECONNECTING }
    }
}

data class ClientConfig(
    val gatewayUrl: String,
    val apiBaseUrl: String,
    val connectTimeoutMs: Int = 10_000,
    val autoReconnect: Boolean = true,
)

File: bindings/ios/swift/AnyChatClient.swift
import Foundation

/// Swift wrapper around the SWIG-generated Objective-C bindings.
/// Provides an async/await API for iOS developers.
public class AnyChatClient {

    private let config: ClientConfig

    public init(config: ClientConfig) {
        self.config = config
    }

    // TODO: delegate to SWIG-generated ObjC wrapper
    public func connect() async throws {
        throw AnyChatError.notImplemented
    }

    public func disconnect() {
        // TODO
    }
}

public struct ClientConfig {
    public let gatewayUrl: String
    public let apiBaseUrl: String
    public var connectTimeoutMs: Int = 10_000
    public var autoReconnect: Bool = true

    public init(gatewayUrl: String, apiBaseUrl: String) {
        self.gatewayUrl = gatewayUrl
        self.apiBaseUrl = apiBaseUrl
    }
}

public enum AnyChatError: Error {
    case notImplemented
    case connectionFailed(String)
    case authFailed(String)
}
